lib/kubernetes-deploy/pod_logger.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class PodLogger
+    def initialize(stream:, logger:, pod_name:, pod_id:)
+      @stream = stream
+      @logger = logger
+      @pod_name = pod_name
+      @pod_id = pod_id
+    end
+
+    def log(kubectl:, deploy_succeeded:, deploy_started_at:, unmanaged:, container_names:)
+      if unmanaged
+        if @stream
+          stream_logs(kubectl, deploy_started_at, unmanaged, container_names)
+        elsif deploy_succeeded
```

Comment:
> is that because if the deplay failed, there can't possibly be any logs? (is that true?)

No, not true. There are three times we might want to output pod logs:
- Inline dump: a script-runner ("unmanaged") pod terminates successfully and we're about to move on with the deploy, e.g. end of Phase 3 [here](https://shipit.shopify.io/shopify/cumulus-cat/production/deploys/624563)
- Inline stream: the new use case in kubernetes-run, where we're waiting on a single pod and want logs to appear asap
- Dump in summary: Either the pod itself or its parent (if it was the selected child) failed, and we want to dump debug information (which includes logs) in the deploy summary section, e.g. the summary section [here](https://shipit.shopify.io/shopify/cumulus-cat/configdemo/deploys/565200)

Reply:
This is the existing behavior. There def can be logs, but the assumption is we want to see logs for unmanaged pods only on success. (In a normal deploy this would be the asset-uploader pod, but I can't speak to why its that way)